include/dr/shp/algorithms/inclusive_scan.hpp

```diff
@@ -38,18 +38,16 @@ void inclusive_scan_impl_(ExecutionPolicy &&policy, R &&r, O &&o,
 
     std::vector<sycl::event> events;
 
-    auto root = shp::devices()[0];
-    shp::device_allocator<T> allocator(shp::context(), root);
+    sycl::queue root = __detail::default_queue();
```

Comment:
For operations like allocation we should actually create the queue on a particular device, so this needs to be reverted.

Reply:
> For operations like allocation we should actually create the queue on a particular device, so this needs to be reverted.

My change did not changed device or context in this place. It was `shp::devices()[0]` and it is `default_queue` which is the same device. Context also is the same. If so - revert fixes nothing. Correct me if I'm wrong.